### ISSUE TITLE
Configuration file and testing

### ISSUE THREAD

**Issue by felopla** [maths/stack_util_maximapool#15]
Hi friends, 

I would like to ask your help to configure MaximaPool. 
In one server, I have installed Moodle, Stack and Maxima 5.36.0. Locally, all work good. The headcheck passed ok. 

In a second server, I installed Maxima 5.36.0 and I followed the steps to install MaximaPool (10 steps), but I have some problems in the configuration file. In the test interface (http://localhost:8080/MaximaPool/MaximaPool) I ran the two tests: 
- The first test ("Run the low-level health-check") starts well but ends after a few seconds with TIMEOUT. Is it correct?
- The second test("Run the high-level health-check"), it gets stuck, only appears "Trying to start a Maxima process". 

And the form test in the main page doesn't work, also it gets stuck.

I attach the configuration file. I don't understand "maxima.path.command" and the "Usable placeholders". Can you explain me what I need to put here, please? 

I really appreciate it any your help.
Thanks in advance.

Kind regards
     Felipe López P.

`### Maxima related commands and parameters`
`maxima.commandline = /usr/local/bin/maxima`
`# a file to load to maxima using load(""); set to false or empty if not needed`
`maxima.load = /opt/moodle/apps/moodle/moodledata/stack/maximalocal.mac`
`# working directory for the process`
`maxima.cwd = .`
`# to identify the state of the process we check the output for specific strings, if you change the` `loading process change these strings to match`
`# Input the load command after this, if defined...`
`maxima.ready.for.load = (%i1)`
`# The process has loaded when this appears.`
`maxima.ready.for.use = Stack-Maxima started`

`### File handling related parameters, these make the difference between shared filesystem` `installations and truly separate servers`
`# Do we handle files?`
`file.handling = true`
`# The command template to pass the current working and output directories to the process, use` `placeholders to mark the paths, depending on your system we may escape \-chars in the path.`
`## Usable placeholders:`
`##  %WORK-DIR% the directory that won't be transferred to the client side`
`##  %WORK-DIR-NE% use this if you do not want \-chars escaped`
`##  %OUTPUT-DIR% the directory that will be transferred to the client side`
`##  %OUTPUT-DIR-NE% use this if you do not want \-chars escaped`
`maxima.path.command = TMP_IMAGE_DIR: "/opt/tomcat/prueba/"; IMAGE_DIR: "/opt/tomcat/prueba2/"; URL_BASE: "%PLOT-URL-BASE%"; `

![1](https://cloud.githubusercontent.com/assets/12087852/9635084/f77bf594-516a-11e5-9523-e93ba6fda991.png)
![2](https://cloud.githubusercontent.com/assets/12087852/9635085/f77ef122-516a-11e5-869d-83fa76e19d24.png)
![3](https://cloud.githubusercontent.com/assets/12087852/9635030/b03b720e-516a-11e5-9b38-faff3f587ae1.png)

**Comment by aharjula** [comment]
Hi,

Well the problem seems to be that you have the pool configured to wait for the string "Stack-Maxima started" but as that string newer appears it fails as it runs out of time. Naturally, the pool has that as the default value to look for but STACK itself has changed that string. So try this:

maxima.ready.for.use = STACK-Maxima started

As to the maxima.path.command parameter the place holders should not be replaced with static values, the pool is meant to run multiple processes in different paths and needs a way to tell STACK-Maxima what paths it is using. For this reason is has these markers that you are supposed to use in a code segment that the pool can inject to maxima. Again there has been some changes to the correct value of this from STACKs side, use this:

maxima.path.command = maxima_tempdir: "%WORK-DIR%/"$ IMAGE_DIR: "%OUTPUT-DIR%/"$ URL_BASE: "%PLOT-URL-BASE%"$

Basically, "TMP_IMAGE_DIR" is now "maxima_tempdir". And if you were to use an older version of STACK you might need to change that back or even define both to get plots working.

Anyway sorry that we have not had the time to keep all the pool default parameters up to date.

Do let us know if those two parameter changes are enough.

-Matti Harjula

**Comment by felopla** [comment]
Dear Matti, 

Thanks a lot. Works like a charm!

:)

Thank you very much. Have a good afternoon.

Kind regards
Felipe López P.
